Automattic/newspack-popups
# feat: render a preview of the dismiss button in the editor

### All Submissions:






### Changes proposed in this Pull Request:



Uses `editor.BlockListBlock` filter to render a preview of the dismiss button text and alignment at the end of the editor block list.

Closes #311.

### How to test the changes in this Pull Request:

1. Check out this branch and run `npm ci && npm run build` (must install new NPM packages before building).
2. Create a new campaign or edit an existing one.
3. Observe that a non-interactive preview of the dismiss button now appears after the last block in the campaign post content. ~**Note:** If testing with a new campaign, the preview will not appear until the content contains at least one block, e.g. after clicking into the block editor area.~

<img width="846" alt="Screen Shot 2020-12-07 at 6 55 30 PM" src="https://user-images.githubusercontent.com/2230142/101428608-4ce5c380-38be-11eb-902e-ee06445c12e3.png">

4. Test that when changing the dismiss button text and alignment, the preview updates itself accordingly.
5. Test that the preview only renders after the last block, even after adding and removing blocks. ~**Note:** On initial page render, the block appender button may appear underneath the preview, but the preview will move itself if more blocks are added to the content.~
6. Test that the preview does not render with any post type other than Campaigns.

### Other information:
